API/dag/README.md

```diff
@@ -38,3 +49,71 @@ If no `callback` is passed, a [promise][] is returned.
 
 - `value` - the value or node that was fetched during the get operation.
 - `remainderPath` - The remainder of the Path that the node was unable to resolve or what was left in a localResolve scenario.
+
+If no `callback` is passed, a [promise][] is returned.
+
+**Example:**
+
+```JavaScript
+```
+
+[A great source of examples can be found in the tests for this API.][examples]
+
+#### `dag.ls`
+
+> Enumerate all the first level link names
+
+##### `Go` **WIP**
+
+##### `JavaScript` - ipfs.dag.ls(cid [, options], callback)
+
+- `cid` - can be one of the following:
+  - a [CID](https://github.com/ipfs/js-cid) instance.
+  - a CID in its String format (i.e: zdpuAkxd9KzGwJFGhymCZRkPCXtBmBW7mB2tTuEH11HLbES9Y)
+  - a CID in its String format concatenated with the path to be resolved
+- `options` - a object that might contain the following values:
+  - `stream` - bool - if set to true, it will return a Node.js Readable Stream.
+  - `pull` - bool - if set to true, it will return a pull-stream.
```

Comment:
@dignifiedquire understood. Right now it is PITA because we have places where 'xStream' is returning a pull-stream, other places where there is an option (really legacy) for `buffer: true` and a lot of internals that are pull-streams and the 'external' (ie js-ipfs) is Node.js Streams for the sake of not breaking API.

I would be down to just sit, define a language for this different options (or through options object or through different methods) and then apply as a golden rule throughout all the modules.